taipy/core/job/job.py

```diff
@@ -308,9 +308,6 @@ def _on_status_change(self, *functions):
         function = functions.pop()
         self._subscribers.append(function)
 
-        if self.status != Status.SUBMITTED:
-            function(self)
-
         if functions:
             self._on_status_change(*functions)
```

Comment:
In my opinion, the name of the function is no longer applicable. Should we rename it to something like `_set_on_status_change_callbacks`?